### PR TITLE
Added the ability to step backwards in tutorial

### PR DESCRIPTION
**Pull request by ePak**
Added the ability to show a previous button on the tooltips such that users could go back to the previous steps. This can be enabled via the `TutorialConfiguration.allowSteppingBackward` option (default: false).

TODO

- [x] update docs
- [ ] update README 
- [ ] update demo

@haabaato Any thoughts?

### CODE REVIEW

File: docs/tutorial.js.html [hunk 1]
```diff
@@ -267,10 +272,10 @@ <h1 class="page-title">Source: tutorial.js</h1>
     let nextStep;
     if (step) {
       if (typeof step === 'number') {
-        if (step &lt; 0 || step >= this.steps.length) {
+        if (step &lt; 1 || step > this.steps.length) {
```
Comment (ePak) [inline docs/tutorial.js.html]: I think using one-indexed step number is more consistent with the `stepNum(step)` function.
Nothing inside Chariot is passing a step number into this function at the moment, but this could be a breaking change for library users.

**Comment by ePak** [comment]
[Chariot enhancement](https://trello.com/c/ZszxywJq/67-chariot-enhancement)

File: docs/tutorial.js.html (hunk 1 above)
Comment (haabaato) [inline docs/tutorial.js.html]: @th11 @stephaniewei Any idea if this would break our usage of chariot in product?

File: stylesheets/example-tooltip.scss [hunk 2]
```diff
@@ -50,18 +50,27 @@ $shadow-color: rgba(31, 28, 28, 0.3);
...
+      border: 1px solid #78A300;
       border-radius: 3px;
       height: 36px;
       padding: 0 20px;
+
+    }
+
+    .btn-next {
+      color: $white;
+      background-color: #78A300;
```
Comment (haabaato) [inline stylesheets/example-tooltip.scss]: Could you add some screenshots of what the tooltip looks like after adding the previous button and these style changes?

Comment (ePak) [inline stylesheets/example-tooltip.scss]: image
Here is a screenshots for now, I will up update the live demo as well if this PR would get merged.

Comment (haabaato) [inline stylesheets/example-tooltip.scss]: Looks good, but let me confirm that my team won't be affected by this.

File: lib/tutorial.js [hunk 3]
```diff
@@ -241,10 +244,10 @@ class Tutorial {
     let nextStep;
     if (step) {
       if (typeof step === 'number') {
-        if (step < 0 || step >= this.steps.length) {
+        if (step < 1 || step > this.steps.length) {
```
Comment (haabaato) [inline lib/tutorial.js]: Let's undo this change.  Our current usage of chariot would break, and we already have the `stepNum() / currentStepNum()` methods to surface the one-indexed (human-friendly) index.

File: lib/step.js [hunk 4]
```diff
@@ -139,20 +139,29 @@ class Step {
...
+      action();
     }).catch(error => {
       console.log(error);
-      this.tutorial.next();
+      action();
     });
   }
 
+  previous() {
+    let step = this.tutorial.stepNum(this) - 1;
```
Comment (ePak) [inline lib/step.js]: I used `stepNum()` instead of `this.index` because I had made `tutorial.next()` accepts a one-indexed step number. So, if I change `tutorial.next()` back to zero-indexed then I could use `this.index`. But either way, the `-1` is still needed to get the index of the previous step.

**Comment by ePak** [comment]
💇‍♂️ @haabaato

File: lib/step.js (hunk 4 above)
Comment (haabaato) [inline lib/step.js]: Oh right, that's obvious now. Thanks for reverting the one-index!

**Comment by haabaato** [comment]
LGTM :+1:
If you have time, could you please update chariot.zendesk.com example as well?

File: example/index.html [hunk 5]
```diff
@@ -65,7 +65,7 @@
 
 var delegate = this;
 
-var chariot = new Chariot({
+var chariotInstance = new chariot({
```
Comment (zendesk-yoshi-sono) [inline example/index.html]: Should class name be kept capitalized?

Comment (ePak) [inline example/index.html]: We could. But this line wasn't working before, I think the class was being exported as `chariot`. It could break something else if they are already using `new chariot()` and we change it.
